gsoap-2.8.17/debian/soapcpp2.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH SOAPCPP2 1 "Juni 27, 2003"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
soapcpp2 \- the gSOAP Stub and Skeleton Compiler for C and C++
.SH SYNOPSIS
\fBsoapcpp2\fR [\fIOPTIONS\fR] \fIINPUT\fR
.SH DESCRIPTION
Please see /usr/share/doc/gsoap/soapdoc2.html for details.
.PP
Create stubs and client and server code from input \fIINPUT\fR or
standard input if \fIINPUT\fR is not specified.
.SH OPTIONS
.TP
\fB\-1\fR
Generate SOAP 1.1 bindings.
.TP
\fB\-2\fR
Generate SOAP 1.2 bindings.
.TP
\fB\-0\fR
Remove SOAP bindings, use REST.
.TP
\fB\-C\fR
Generate client-side code only.
.TP
\fB\-S\fR
Generate server-side code only.
.TP
\fB\-T\fR
Generate server-side auto-test code.
.TP
\fB\-L\fR
Do not generate \fBsoapClientLib\fR/\fBsoapServerLib\fR.
.TP
\fB\-a\fR
Use \fBSOAPAction\fR with WS-Addressing to invoke server-side operations.
.TP
\fB\-A\fR
Require \fBSOAPAction\fR to invoke server-side operations.
.TP
\fB\-b\fR
Serialize byte arrays char[N] as string.
.TP
\fB\-c\fR
Generate C source code.
.TP
\fB\-d\fIpath\fR
Use \fIpath\fR to save files.
.TP
\fB\-e\fR
Generate SOAP RPC encoding style bindings.
.TP
\fB\-f\fIN\fR
File split of \fIN\fR XML serializer implementations per file (N>=10).
.TP
\fB\-h\fR
Display help info.
.TP
\fB\-I\fIpath\fR
use \fIpath\fR(s) for \fB#import\fR (paths separated with ':').
.TP
\fB\-i\fR
Generate C++ service proxies and objects inherited from \fBsoap\fR struct.
.TP
\fB\-j\fR
Generate C++ service proxies and objects that share a \fBsoap\fR struct.
.TP
\fB\-k\fR
Generate data structure walkers (experimental).
.TP
\fB\-l\fR
Generate linkable modules (experimental).
.TP
\fB\-m\fR
Generate Matlab(tm) code for MEX compiler.
.TP
\fB\-n\fR
Use service name to rename service functions and namespace table.
.TP
\fB\-p\fIname\fR
Save files with new prefix \fIname\fR instead of \fBsoap\fR.
.TP
\fB\-Q\fIname\fR
Use \fIname\fR as the C++ namespace for decls, including custom serializers.
.TP
\fB\-q\fIname\fR
Use \fIname\fR as the C++ namespace for decls, excluding custom serializers.
.TP
\fB\-s\fR
Generate deserialization code with strict XML validation checks.
.TP
\fB\-t\fR
Generate code for fully \fBxsi:type\fR typed SOAP/XML messaging.
.TP
\fB\-u\fR
Uncomment comments in WSDL/schema output by suppressing XML comments.
.TP
\fB\-v\fR
Display version info.
.TP
\fB\-w\fR
Do not generate WSDL and schema files.
.TP
\fB\-x\fR
Do not generate sample XML message files.
.TP
\fB\-y\fR
Include C/C++ type access information in sample XML messages.
.TP
\fB\-z1\fR
Generate deprecated old-style C++ service proxies and objects.
.SH SEE ALSO
.BR wsdl2h (1).
.SH AUTHOR
This manual page was written by Thomas Wana <greuff@debian.org>,
for the Debian project (but may be used by others).


noflushd/NEWS
* noflushd 2.7.5 (09. May 2005)

  - Fix handling of raid and similar meta devices on 2.6 kernels.

* noflushd 2.7.4 (08. Aug 2004)

  - Fix lousy spindown times on 2.6 kernels.
  - Fix SCSI support on 2.6 kernels.
  - Fix pid detection in generic startup script.
  - Document quirks with hotplugged drives.

* noflushd 2.7.3 (09. Apr 2004)

  - Fix race condition that might leave kupdated/pdflushd disabled
    aftern noflushd has terminated.
  - Fix spindown when all disks are auto-detected.

* noflushd 2.7.2 (16. Mar 2004)

  - A change in 2.7.1 erroneously flushed the buffer cache before
    spindown, causing frequent spinups. This change has been reverted.
    
* noflushd 2.7.1 (07. Mar 2004)

  - Fix immediate spinup on 2.6 kernels.
  - Elevate SCSI support from compile-time to command line option.
  - Bug fixes in the disk statistics parser for 2.6 kernels.
  - Ship kernel patch to track down which processes cause spinups.

* noflushd 2.7 (20. Jan 2004)

  - Support Linux kernel 2.6.x.
  - Default to monitorring all disks unless specific disks are given
    on the command line. This change should yield a sane default
    behaviour for a majority of users without requiring explicit
    configuration.

* noflushd 2.6.3 (13. May 2002)

  - Bugfix for arm and other archs with weird /proc/partitions layout.

* noflushd 2.6.2 (20. Apr 2002)

  - Add IDE raid support.
  - Add timeout logging script.

* noflushd 2.6.1 (03. Feb. 2002)

  - Fix BUG when using noflushd with disks on second IDE channel on 2.4
    kernels.

* noflushd 2.6 (22. Jan 2002)

  - Document noflushd's suckage with any journaling fs, including ext3.
  - New feature prevents spindown if machine is in interactive use.
  - Add hack fix some '0 spindown time' uglies.

* noflushd 2.5 (06. Aug 2001)

  - Make sure hotplugged disks get synced properly.
  - Only sync present partitions. (Fixes high CPU usage bug on devfs
    systems.)
  - Generic startup skript works on latest 2.4 kernels.

* noflushd 2.4 (13. Mar 2001)

  - Timeout lists are back to a working state.
  - Support very latest 2.4 kernels.

* noflushd 2.3 (12. Jan 2001)

  - Support latest 2.4 kernels. (Leaves kernels 2.3.99pre8 to 2.4.0-test8
    unsupported.)
  - Fix multi-disk support with long bdflush timeouts.

* noflushd 2.2 (22. Dec 2000)

  - Fixed bug in spindown code. Now actually spins down. Duh!

* noflushd 2.1 (20. Dec 2000)

  - Disks get synced prior to spindown.
  - Improved devfs support.

* noflushd 2.0 (14. Nov 2000)

  - noflushd now comes with its own syncer that takes over the task from
    kupdate when the first read-write mounted disk is spun down. This
    enhances spindown times in a multidisk environment at the expense of
    less smooth and more bursty background flushing when the noflushd 
    syncer is in use.

  - Conflicting entries in 2.2-style /proc/stat (e.g. /dev/sda and /dev/hda)
    won't cause noflushd to give up anymore. Instead, noflushd will treat
    both disks pretty much like one, i.e. they're either both up or spun
    down, no matter if activity only goes to one of them.

  - Compilation environment switched to automake/autoconf.

  - When installing via 'make install', noflushd is no longer added to
    any runlevel. 

  - noflushd moved to SourceForge. The project homepage is located at
    http://noflushd.sourceforge.net, a mailing list is available via
    noflushd-devel@lists.sourceforge.net. Visit the homepage for 
    subscription information.



gsoap-2.8.17/gsoap/src/soapcpp2.c
/*
	soapcpp2.c

	Main compiler and code generator batch program.

gSOAP XML Web services tools
Copyright (C) 2000-2013, Robert van Engelen, Genivia Inc. All Rights Reserved.
This part of the software is released under one of the following licenses:
GPL or Genivia's license for commercial use.
--------------------------------------------------------------------------------
GPL license.

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 2 of the License, or (at your option) any later
version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with
this program; if not, write to the Free Software Foundation, Inc., 59 Temple
Place, Suite 330, Boston, MA 02111-1307 USA

Author contact information:
engelen@genivia.com / engelen@acm.org

This program is released under the GPL with the additional exemption that
compiling, linking, and/or using OpenSSL is allowed.
--------------------------------------------------------------------------------
A commercial use license is available from Genivia, Inc., contact@genivia.com
--------------------------------------------------------------------------------
*/

#include "soapcpp2.h"

#ifndef SOAPCPP2_IMPORT_PATH
#define SOAPCPP2_IMPORT_PATH (NULL)
#endif

extern void init(void);
extern int yyparse(void);
extern FILE *yyin;

extern char *ns_cname(char*, char*);

FILE *fmsg;		/* fd to flush compiler messages */

int vflag = 0;		/* SOAP version, -1=no SOAP, 0=not set, 1=1.1, 2=1.2 */
int wflag = 0;		/* when set, don't generate WSDL and schema files */
int Cflag = 0;		/* when set, generate only files for clients */
int cflag = 0;		/* when set, generate files with .c extension */
int aflag = 0;		/* when set, use value of SOAP Action to dispatch method at server side */
int Aflag = 0;		/* when set, require SOAP Action to dispatch method at server side */
int bflag = 0;		/* when set, serialize byte arrays char[N] as string */
int eflag = 0;		/* when set, use SOAP RPC encoding by default */
unsigned long fflag = 0;/* multi-file split for each bundle of -fN defs */
int iflag = 0;		/* when set, generate new style proxy/object classes inherited from soap struct */
int jflag = 0;		/* when set, generate new style proxy/object classes */
int kflag = 0;		/* when set, generate data traversal/walker routines */
int mflag = 0;		/* when set, generate code that requires array/binary classes to explicitly remove malloced array */
int nflag = 0;		/* when set, names the namespaces global struct '%NAME%_namespaces */
int lflag = 0;		/* when set, create library */
int Lflag = 0;		/* when set, don't generate soapClientLib/soapServerLib */
int Qflag = 0;		/* when set, use C++ namespaces for custom serializers */
int sflag = 0;		/* when set, generate strict validation checks */
int Sflag = 0;		/* when set, generate only files for servers */
int Tflag = 0;		/* when set, generates server auto-test code */
int tflag = 0;		/* when set, generates typed messsages (with xsi:type attributes) */
int uflag = 0;		/* when set, uncomment WSDL and schema output */
int xflag = 0;		/* when set, don't generate sample XML message files */
int yflag = 0;		/* when set, add C/C++ info in sample XML messages */
int zflag = 0;		/* when set, use backward compatibility option */

int stop_flag = 0;

char dirpath[1024];	/* directory path for generated source files */
char *prefix = "soap";	/* file name prefix for generated source files */
char filename[1024];	/* current file name */
char *importpath = NULL;
char *defimportpath = SOAPCPP2_IMPORT_PATH; /* default file import path */

/*
IMPORTANT:

The terms and conditions of use of this software do not allow for the removal
of the copyright notice from the main program for visual display. For
integrations with other software, a similar copyright notice must be produced
that is visible to users of the software.
*/

int
main(int argc, char **argv)
{	int i, g;
	char *a, *s;
	fmsg = stderr;
	strcpy(filename, "<stdin>");
	for (i = 1; i < argc; i++)
	{	a = argv[i];
		if (*a == '-'
#ifdef WIN32
		 || *a == '/'
#endif
		)
		{	g = 1;
			while (g && *++a)
				switch (*a)
				{	case 'C':
						Cflag = 1;
						if (Sflag)
            					  fprintf(stderr, "soapcpp2: using both options -C and -S omits client/server code\n");
						break;
					case 'c':
						cflag = 1;
						break;
					case 'd':
						a++;
						g = 0;
						if (*a)
							strcpy(dirpath, a);
						else if (i < argc && argv[++i])
							strcpy(dirpath, argv[i]);
						else
							execerror("Option -d requires a directory path");
						if (*dirpath && dirpath[strlen(dirpath)-1] != '/' && dirpath[strlen(dirpath)-1] != '\\')
							strcat(dirpath, SOAP_PATHCAT);
						break;
					case 'e':
						eflag = 1;
						break;
					case 'f':
						a++;
						g = 0;
						if (*a)
							fflag = strtoul(a, NULL, 10);
						else if (i < argc && argv[++i])
							fflag = strtoul(argv[i], NULL, 10);
						if (!fflag)
							execerror("Option -f requires a value");
						if (fflag < 10)
						  fflag = 10;
						break;
					case 'a':
						aflag = 1;
						break;
					case 'A':
						aflag = 1;
						Aflag = 1;
						break;
					case 'b':
						bflag = 1;
						break;
					case '?':
					case 'h':
						fprintf(stderr, "Usage: soapcpp2 [-0|-1|-2] [-C|-S] [-T] [-L] [-a] [-A] [-b] [-c] [-d path] [-e] [-f N] [-h] [-i] [-I path"SOAP_PATHSEP"path"SOAP_PATHSEP"...] [-k] [-l] [-m] [-n] [-p name] [-s] [-t] [-u] [-v] [-w] [-x] [-y] [-z#] [infile]\n\n");
						fprintf(stderr, "\
-1      generate SOAP 1.1 bindings\n\
-2      generate SOAP 1.2 bindings\n\
-0      remove SOAP bindings, use REST\n\
-C	generate client-side code only\n\
-S	generate server-side code only\n\
-T	generate server auto-test code\n\
-L	don't generate soapClientLib/soapServerLib\n\
-a	use SOAPAction with WS-Addressing to invoke server-side operations\n\
-A	require SOAPAction to invoke server-side operations\n\
-b	serialize byte arrays char[N] as string\n\
-c      generate C source code\n\
-dpath  use path to save files\n\
-e	generate SOAP RPC encoding style bindings\n\
-fN	file split of N XML serializer implementations per file (N>=10)\n\
-h	display help info\n\
-Ipath  use path(s) for #import (paths separated with '"SOAP_PATHSEP"')\n\
-i      generate C++ service proxies and objects inherited from soap struct\n\
-j      generate C++ service proxies and objects that share a soap struct\n\
-k      generate data structure walkers (experimental)\n\
-l      generate linkable modules (experimental)\n\
-m      generate Matlab(tm) code for MEX compiler\n\
-n      use service name to rename service functions and namespace table\n\
-pname  save files with new prefix name instead of 'soap'\n\
-Qname  use name as the C++ namespace for decls, including custom serializers\n\
-qname  use name as the C++ namespace for decls, excluding custom serializers\n\
-s      generate deserialization code with strict XML validation checks\n\
-t      generate code for fully xsi:type typed SOAP/XML messaging\n\
-u	uncomment comments in WSDL/schema output by suppressing XML comments\n\
-v	display version info\n\
-w	don't generate WSDL and schema files\n\
-x	don't generate sample XML message files\n\
-y	include C/C++ type access information in sample XML messages\n\
-z1	generate deprecated old-style C++ service proxies and objects\n\
infile	header file to parse (or stdin)\n\
\n");
						exit(0);
					case 'I':
						a++;
						g = 0;
						s = NULL;
						if (*a)
							s = a;
						else if (i < argc && argv[++i])
							s = argv[i];
						else
							execerror("Option -I requires an import path");
						if (importpath && s)
						{	char *t	= emalloc(strlen(importpath) + strlen(s) + 2);
							strcpy(t, importpath);
							strcat(t, SOAP_PATHSEP);
							strcat(t, s);
							importpath = t;
						}
						else
							importpath = s;
						break;
					case 'i':
						iflag = 1;
						break;
					case 'j':
						jflag = 1;
						break;
					case 'k':
						kflag = 1;
						break;
					case 'm':
						mflag = 1;
						break;
					case 'n':
						nflag = 1;
						break;
					case 'l':
						lflag = 1;
						break;
					case 'L':
						Lflag = 1;
						break;
					case 's':
						sflag = 1;
						break;
					case 'S':
						Sflag = 1;
						if (Cflag)
            					  fprintf(stderr, "soapcpp2: using both options -C and -S omits client/server code\n");
						break;
					case 'T':
						Tflag = 1;
						break;
					case 't':
						tflag = 1;
						break;
					case 'u':
						uflag = 1;
						break;
					case 'w':
						wflag = 1;
						break;
					case 'x':
						xflag = 1;
						break;
					case 'y':
						yflag = 1;
						break;
					case 'p':
						a++;
						g = 0;
						if (*a)
							prefix = ns_cname(a, NULL);
						else if (i < argc && argv[++i])
							prefix = ns_cname(argv[i], NULL);
						else
							execerror("Option -p requires an output file name prefix");
						break;
					case 'Q':
						Qflag = 1;
						/* fall through */
					case 'q':
						a++;
						g = 0;
						if (*a)
							namespaceid = ns_cname(a, NULL);
						else if (i < argc && argv[++i])
							namespaceid = ns_cname(argv[i], NULL);
						else
							execerror("Option -q requires a namespace name");
						break;
					case '0':
						vflag = -1;
						break;
					case '1':
						vflag = 1;
						envURI = "http://schemas.xmlsoap.org/soap/envelope/";
						encURI = "http://schemas.xmlsoap.org/soap/encoding/";
						break;
					case '2':
						vflag = 2;
						envURI = "http://www.w3.org/2003/05/soap-envelope";
						encURI = "http://www.w3.org/2003/05/soap-encoding";
						rpcURI = "http://www.w3.org/2003/05/soap-rpc";
						break;
					case 'v':
						stop_flag = 1;
						break;
					case 'z':
						a++;
						g = 0;
						if (*a)
							zflag = *a - '0';
						else if (i < argc && argv[++i])
							zflag = *argv[i] - '0';
						else
							execerror("Option -z requires a digit");
						break;
					default:
            					fprintf(stderr, "soapcpp2: Unknown option %s\n", a);
            					exit(1);
				}
		}
		else if (!(yyin = fopen(argv[i], "r")))
		{	sprintf(errbuf, "Cannot open file \"%s\" for reading", argv[i]);
			execerror(errbuf);
		}
		else
			strcpy(filename, argv[i]);
	}

	if (importpath && defimportpath)
	{	char *t	= emalloc(strlen(importpath) + strlen(defimportpath) + 2);
		strcpy(t, importpath);
		strcat(t, SOAP_PATHSEP);
		strcat(t, defimportpath);
		importpath = t;
	}
	else
		importpath = defimportpath;

	fprintf(fmsg, "\n**  The gSOAP code generator for C and C++, soapcpp2 release "VERSION"\n**  Copyright (C) 2000-2013, Robert van Engelen, Genivia Inc.\n**  All Rights Reserved. This product is provided \"as is\", without any warranty.\n**  The soapcpp2 tool is released under one of the following licenses:\n**  GPL or the commercial license by Genivia Inc.\n\n");
	if (stop_flag)
	  exit(0);
	init();
	if (yyparse())
		synerror("skipping the remaining part of the input");
	return errstat();
}
